mdapi/src/classes/ExpenseReportSharingTest.cls

```diff
@@ -0,0 +1,68 @@
+@isTest
+private class  ExpenseReportSharingTest {
+
+    @isTest
+    static void whenProfileIsFinanceCanGetAllExpenseReports() {
+
+        Expense_Report__c anothers = new Expense_Report__c(
+            Name = 'Show me their money',
+            Amount__c = 100
+        );
+        insert anothers;
+        
+        User user = [SELECT Id
+                    FROM User
+                    WHERE Profile.Name = 'Finance'
+                    LIMIT 1];
+        System.assert(user != null, 'User Exists');
+
+        System.runAs(user) {
+            Expense_Report__c mine = new Expense_Report__c(
+                Name = 'Show me my money',
+                Amount__c = 500
+            );
+            insert mine;
+
+            Test.startTest();
+            List<Expense_Report__c> saved = [
+                SELECT Id
+                FROM Expense_Report__c
+            ];
+            Test.stopTest();
+
+            System.assertEquals(2, saved.size(), 'can see all the money');
+        }
+    }
+    @isTest
+    static void whenProfileIsConsultantCantGetAllExpenseReports() {
+        
+        Expense_Report__c anothers = new Expense_Report__c(
+            Name = 'Show me their money',
+            Amount__c = 100
+        );
+        insert anothers;
+        
+        User user = [SELECT Id
+                    FROM User
+                    WHERE Profile.Name = 'Consultant'
+                    LIMIT 1];
+        System.assert(user != null, 'User Exists');
+
+        System.runAs(user) {
+            Expense_Report__c mine = new Expense_Report__c(
+                Name = 'Show me my money',
+                Amount__c = 500
+            );
+            insert mine;
+
+            Test.startTest();
+            List<Expense_Report__c> saved = [
+                SELECT Id
+                FROM Expense_Report__c
+            ];
+            Test.stopTest();
+
+            System.assert(!saved.contains(anothers) , 'can only see my money');
```

Comment:
In the case of apex Lists and objects, I wouldn't recommend using `contains` since the matching criteria might be ambiguous and could result in a false positive assertion.

Try asserting size to 1 and asserting that element at `saved[0].id` is the same id as `mine.id`

Reply:
@kikemoto would `mine.id` be like ownerId? Because my expense report object only has createdById, which isn't necessarily the owner, right? I assumed owner id was a field that gets automatic created, but I don't see it.